### PR TITLE
Add pmon daemons python3 build support

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
**- Why I did it**

python2 is end of life and SONiC is going to support python3. This PR is going to support:

1. Build pmon daemons with python3
2. Install and run python3 version pmon daemons

**- How I did it**

1. Change pmon daemons make files to build bothe python2 and python3 whl
2. Change docker-platform-monitor make files to install both python2 and python3 whl
3. Change pmon docker startup files to start pmon daemons according to the supported platform API version

**- How to verify it**

1. Manual test to make sure all daemons are up and no error in syslog
2. Run platform regressions on a few platforms

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: rules/docker-platform-monitor.mk [hunk 1]
```diff
@@ -6,23 +6,29 @@ DOCKER_PLATFORM_MONITOR_DBG = $(DOCKER_PLATFORM_MONITOR_STEM)-$(DBG_IMAGE_MARK).
...
+$(DOCKER_PLATFORM_MONITOR)_PYTHON_WHEELS += $(SONIC_XCVRD_PY2)
 ifeq ($(PDDF_SUPPORT),y)
 $(DOCKER_PLATFORM_MONITOR)_PYTHON_WHEELS += $(PDDF_PLATFORM_API_BASE_PY2)
 endif
+$(DOCKER_PLATFORM_MONITOR)_PYTHON_WHEELS += $(SONIC_LEDD_PY3)
+$(DOCKER_PLATFORM_MONITOR)_PYTHON_WHEELS += $(SONIC_PCIED_PY3)
+$(DOCKER_PLATFORM_MONITOR)_PYTHON_WHEELS += $(SONIC_PSUD_PY3)
+$(DOCKER_PLATFORM_MONITOR)_PYTHON_WHEELS += $(SONIC_SYSEEPROMD_PY3)
+$(DOCKER_PLATFORM_MONITOR)_PYTHON_WHEELS += $(SONIC_THERMALCTLD_PY3)
+$(DOCKER_PLATFORM_MONITOR)_PYTHON_WHEELS += $(SONIC_XCVRD_PY3)
```
Comment (jleveque) [inline rules/docker-platform-monitor.mk:31]: Suggest moving this section above line 23

**Review by jleveque — CHANGES_REQUESTED** [review]

**Comment by Junchao-Mellanox** [comment]
The build issue will be fixed after merging https://github.com/Azure/sonic-platform-common/pull/154 and moving submodule pointer.

**Review by jleveque — DISMISSED** [review]

**Comment by jleveque** [comment]
@Junchao-Mellanox: Please fix new conflicts.

**Comment by Junchao-Mellanox** [comment]
> @Junchao-Mellanox: Please fix new conflicts.

Done

**Comment by jleveque** [comment]
Retest vs please

**Comment by jleveque** [comment]
Build failures in unit tests:

```
17:23:50  _____________________ ERROR collecting tests/test_xcvrd.py _____________________
17:23:50  tests/test_xcvrd.py:36: in <module>
17:23:50      class TestXcvrdScript(object):
17:23:50  tests/test_xcvrd.py:41: in TestXcvrdScript
17:23:50      @patch('xcvrd.xcvrd.logical_port_name_to_physical_port_list', MagicMock(return_value=[0]))
17:23:50  E   NameError: name 'patch' is not defined
```

Please check.

**Comment by Junchao-Mellanox** [comment]
> 17:23:50  _____________________ ERROR collecting tests/test_xcvrd.py _____________________
> 17:23:50  tests/test_xcvrd.py:36: in <module>
> 17:23:50      class TestXcvrdScript(object):
> 17:23:50  tests/test_xcvrd.py:41: in TestXcvrdScript
> 17:23:50      @patch('xcvrd.xcvrd.logical_port_name_to_physical_port_list', MagicMock(return_value=[0]))
> 17:23:50  E   NameError: name 'patch' is not defined

Hi @jleveque , this issue is introduced by commit:

```
commit b0be7ca52868064fb4e2b5ef9d44c774fdd87ade
Author: vdahiya12 <67608553+vdahiya12@users.noreply.github.com>
Date:   Tue Dec 15 04:10:34 2020 -0800

    [xcvrd] add unit test infrastructure and unit tests for xcvrd (#133)

    Summary:
    This PR provides the necessary infrastructure to add pytest support and integration in sonic-xcvrd submodule.
    This PR also adds unit tests for xcvrd daemon.
    What is the motivation for this PR?
    To add the pytest unittest support in sonic-platform-daemon, sonix-xcvrd daemon as well as add some unit tests
    Signed-off-by: vaibhav-dahiya <vdahiya@microsoft.com>
```

The error is introduced by line https://github.com/Azure/sonic-platform-daemons/blob/b674dffb717968b306b2bb967b0cbce74661a45f/sonic-xcvrd/tests/test_xcvrd.py#L9 . In this line it should import the package patch like:

```python
if sys.version_info >= (3, 3):
    from unittest.mock import MagicMock, patch
```

I have created a PR to fix this.

**Comment by jleveque** [comment]
@Junchao-Mellanox: Thank you for the fix! Can you please update the submodule?

**Comment by Junchao-Mellanox** [comment]
Sure

**Comment by Junchao-Mellanox** [comment]
Hi @jleveque , please see https://github.com/Azure/sonic-buildimage/pull/6247

**Comment by jleveque** [comment]
> Hi @jleveque , please see #6247

Merged. Thanks again!

**Comment by jleveque** [comment]
Retest this please

**Review by jleveque — APPROVED** [review]

**Pull request merged**